### ISSUE TITLE
Configure geo-redundant options for per-state storage accounts

### ISSUE THREAD

**Issue by ryanhofdotgov** [18F/piipan#891]
- [ ] Evaluate options: https://docs.microsoft.com/en-us/azure/storage/common/storage-disaster-recovery-guidance
- [ ] Allow redundancy setting to be configurable per environment; e.g., we don't need redundancy at all for dev/test environments

**Comment by edwintorres** [comment]
In the process of completing this task, there are several findings. We're putting this task in suspense. 

To change the redundancy configuration to "Zone Redundant" there is the need to change the region to westus2 because the Zone Redundant is not available in westus. When this was done, there was an error with the IaC code. The create-resource.bash is using the activity log, and when using the variable $LOCATION the following code was pointing to westus2 but the activity log is on westus so there was an error.

```
  # Send Policy events from subscription's activity log to event hub
  az deployment sub create \
    --name activity-log-diagnostics \
    --location "$LOCATION" \
    --template-file ./arm-templates/activity-log.json \
    --parameters \
      eventHubName="$EVENT_HUB_NAME" \
      coreResourceGroup="$RESOURCE_GROUP"
```

When testing, the location on this code was hardcoded to westus and the deployment of the Azure Storage to westus2 was successful with the zone redundant correctly configured. See [this branch](https://github.com/18F/piipan/compare/eth/change-azure-storage-configuration?expand=1).

It is possible to try to recreate all the resources on westus2 but there is an [unresolved issue with new resources created after October 2021](https://github.com/18F/piipan/issues/2289). As the need for redundancy on Azure Storage is not critical we're putting this task in suspense for the moment.

**Comment by kingcomma** [comment]
Thanks for this write up! Since the deployment you quoted is a subscription-level deployment (`az deployment sub`), it appears you ran into [the issue described here](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/deploy-to-subscription?tabs=azure-cli#deployment-location-and-name):

> For each deployment name, the location is immutable. You can't create a deployment in one location when there's an existing deployment with the same name in a different location.

After this is unblocked by the fix for #2289, does it seem like a possible solution is to include the location in the deployment name so we aren't trying to re-use the same name across locations? E.g., `--name activity-log-diagnostics-$LOCATION`.

**Comment by edwintorres** [comment]
I updated [the branch](https://github.com/18F/piipan/compare/eth/change-azure-storage-configuration) with the fix to the activity log name.
